visearch-android/src/main/java/com/visenze/visearch/android/ProductSearch.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+
+    private void addAnalyticsParams(ProductBaseSearchParams searchParams) {
+        if (searchParams == null) return;
+
+        SessionManager sessionManager = visenzeAnalytics.getSessionManager();
+        DataCollection dataCollection = visenzeAnalytics.getDataCollection();
+
+
+
+        if (searchParams.getVaUid() == null) {
+            searchParams.setVaUid(sessionManager.getUid());
+        }
+
+        if (searchParams.getVaSid() == null) {
+            searchParams.setVaSid(sessionManager.getSessionId());
+        }
+
+        if (searchParams.getVaAppBundleId() == null) {
+            searchParams.setVaAppBundleId(dataCollection.getAppId());
+        }
+
+        if (searchParams.getVaAppName() == null) {
+            searchParams.setVaAppName(dataCollection.getAppName());
+        }
+
+        if (searchParams.getVaAppVersion() == null) {
+            searchParams.setVaAppVersion(dataCollection.getAppVersion());
+        }
+
+        if (searchParams.getVaDeviceBrand() == null) {
+            searchParams.setVaDeviceBrand(dataCollection.getDeviceBrand());
+        }
+
+        if (searchParams.getVaDeviceModel() == null) {
+            searchParams.setVaDeviceModel(dataCollection.getDeviceModel());
+        }
+
+        if (searchParams.getVaOs() == null) {
+            searchParams.setVaOs(dataCollection.getOs());
+        }
+
+        if (searchParams.getVaOsv() == null) {
+            searchParams.setVaOsv(dataCollection.getOsv());
+        }
+    }
+
+
+
+    public static class Builder {
+        private String mAppKey;
+        private String mPlacement;
+        private String searchApiEndPoint;
+        private String userAgent;
+        private String uid;
+
+        public Builder(String appKey, String placement) {
+            mAppKey = appKey;
+            mPlacement = placement;
+            searchApiEndPoint = SEARCH_URL;
+            userAgent = USER_AGENT + "/" + BuildConfig.VERSION_NAME;
+        }
+
+        public Builder(URL endPoint, String appKey, String placement) {
+            this(appKey, placement);
+            searchApiEndPoint = endPoint.toString();
+        }
+
+
+        public Builder setApiEndPoint(URL endPoint) {
+            searchApiEndPoint = endPoint.toString();
+            return this;
+        }
+
+        public Builder setUserAgent(String userAgent) {
+            this.userAgent = userAgent;
+            return this;
+        }
+
+        public Builder setUid(String uid) {
+            this.uid = uid;
+            return this;
+        }
+
+        public ProductSearch build(Context context) {
+
+            ProductSearch productSearch = new ProductSearch(context,
+                    mAppKey, mPlacement,
+                    searchApiEndPoint,
+                    userAgent, uid);
+
+            return productSearch;
+        }
+    }
+
+    public static interface ResultListener {
+        public void onSearchResult(final ProductResponse response, String errorMsg);
```

Comment:
can add errorCode as well. Since our API is returning this.

Reply:
oh this is network error right?

Reply:
changed already